### PR TITLE
Attach the gestures to the correct (container or platform) view

### LINKED ISSUE #8330: TapGestureRecognizer not working on a Frame in a grid on iOS or Mac
### Description

TapGestureRecognizer not working on a Frame in a grid.

### Steps to Reproduce

Try the following code:

```cs
var frame = new Frame();
frame.GestureRecognizers.Add(new TapGestureRecognizer() { Command = new Command(() => Debug.WriteLine("dfg")) });
```

Add the frame to a grid, run iOS or Mac and attempt to tap/click on it.
[EDIT], It might be more complex, it needs to be in a scrollview that spans all columns in the grid and in a custom layout control. Will put together an example when I get time.
[EDIT 2] Ok, nothing to do with my custom control. A grid, containing scrollview that spans columns, containing a frame that has a tap gesture recogniser added will show the problem.

This is using VSforMac, so the .net version is 6.0.301.

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, macOS

### Affected platform versions

iOS 15

### Did you find any workaround?

Nope

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

In Xamarin.Forms, the gestures were attached to the renderer not the actual view. This fixes #8330 because the FrameRenderer code was designed to work with gestures on the render.

In this PR I do 2 things:
 1. `_platformView = _handler.ToPlatform();` instead of `_platformView = _handler.PlatformView` to ensure that we use the container view / renderer.
 2. Make sure to use the `_platformView` when doing checks because that is what the gesture is attached to.

### Issues Fixed

Fixes #8330

### Notes

Fixing the gestures like this causes this: https://github.com/dotnet/maui/issues/12116

But this is already an issue on Android and Windows... This is a separate issue, but depending on what we decide there determines how we should fix this....

### CODE REVIEW

**Comment by mattleibow** [comment]
How do we even write tests for this...

File: src/Core/tests/DeviceTests.Shared/HandlerTests/HandlerTestBase.cs [hunk 1]
```diff
@@ -78,8 +78,8 @@ protected void InitializeViewHandler(IElement element, IElementHandler handler,
 		{
 			mauiContext ??= MauiContext;
 			handler.SetMauiContext(mauiContext);
-			handler.SetVirtualView(element);
 			element.Handler = handler;
+			handler.SetVirtualView(element);
```
Comment (mattleibow) [inline src/Core/tests/DeviceTests.Shared/HandlerTests/HandlerTestBase.cs:82]: @PureWeen I spotted this. In the real code, we set `Handler` first. It appears that `SetVirtualView` also sets things, but those need to run later? Not quite sure how this is as both set the Handler. But it resulted in the OnHandlerChangedCore running before the container views were created.

**Comment by Redth** [comment]
Will this also help with https://github.com/dotnet/maui/issues/7121 ?

**Comment by mattleibow** [comment]
> Will this also help with #7121 ?

Not sure, this PR just fixes iOS. But let me confirm.

**Comment by mattleibow** [comment]
> > Will this also help with #7121 ?
> 
> Not sure, this PR just fixes iOS. But let me confirm.

It does not, but I think I found the cause and added a comment.

File: src/Controls/src/Core/Platform/GestureManager/GestureManager.iOS.cs [hunk 2]
```diff
@@ -77,24 +88,22 @@ public void Dispose()
...
 				{
-					_platformView.RemoveInteraction(interaction);
+					PlatformView.RemoveInteraction(interaction);
 				}
 			}
 			_gestureRecognizers.Clear();
 
 			Disconnect();
-
-			_platformView = null;
```
Comment (PureWeen) [inline src/Controls/src/Core/Platform/GestureManager/GestureManager.iOS.cs:97]: Should we still set the weakreference to null here? or initialize it with a null target? So we stay consistent?

Comment (mattleibow) [reply src/Controls/src/Core/Platform/GestureManager/GestureManager.iOS.cs]: done!

**Review by PureWeen — APPROVED** [review]

**Pull request merged**